pkg/kubelet/nodeconfig/controller.go

```diff
@@ -0,0 +1,369 @@
+			return
+		}
+		cc.setConfigOK(curDefaultMessage, curDefaultOKReason, apiv1.ConditionTrue)
+		finalConfig = cc.defaultConfig
+		return
+	} // Assert: we will not use the init or default configurations, unless we roll back to lkg; curUID is a real UID
+
+	// check whether the current config is marked bad
+	if bad, entry := cc.isBadConfig(curUID); bad {
+		infof("current config %q was marked bad for reason %q at time %q", curUID, entry.Reason, entry.Time)
+		finalConfig = cc.lkgRollback(entry.Reason, apiv1.ConditionFalse)
+		return
+	}
+
+	// TODO(mtaufen): consider re-verifying integrity and re-attempting download when a load/verify/parse/validate
+	// error happens outside trial period, we already made it past the trial so it's probably filesystem corruption
+	// or something else scary
+
+	// load the current config
+	toVerify, err := cc.loadCheckpoint(curSymlink)
+	if err != nil {
+		// TODO(mtaufen): rollback and mark bad for now, but this could reasonably be handled by re-attempting a download,
+		// it probably indicates some sort of corruption
+		finalConfig = cc.badRollback(curUID, fmt.Sprintf(curFailLoadReasonFmt, curUID), fmt.Sprintf("error: %v", err))
+		return
+	}
+
+	// verify the integrity of the configuration we just loaded
+	toParse, err := toVerify.verify()
+	if err != nil {
+		finalConfig = cc.badRollback(curUID, fmt.Sprintf(curFailVerifyReasonFmt, curUID), fmt.Sprintf("error: %v", err))
+		return
+	}
+
+	// parse the configuration we just loaded into a KubeletConfiguration
+	cur, err := toParse.parse()
+	if err != nil {
+		finalConfig = cc.badRollback(curUID, fmt.Sprintf(curFailParseReasonFmt, curUID), fmt.Sprintf("error: %v", err))
+		return
+	}
+
+	// validate current config
+	if err := validateConfig(cur); err != nil {
+		finalConfig = cc.badRollback(curUID, fmt.Sprintf(curFailValidateReasonFmt, curUID), fmt.Sprintf("error: %v", err))
+		return
+	}
+
+	// check for crash loops if we're still in the trial period
+	if cc.curInTrial(cur.ConfigTrialDuration.Duration) {
+		if cc.crashLooping(*cur.CrashLoopThreshold) {
+			finalConfig = cc.badRollback(curUID, fmt.Sprintf(curFailCrashLoopReasonFmt, curUID), "")
+			return
+		}
+	} else if !cc.curIsLkg() {
+		// when the trial period is over, the current config becomes the last-known-good
+		cc.setCurAsLkg()
+	}
+
+	// update the status to note that we will use the current config
+	cc.setConfigOK(fmt.Sprintf(curRemoteMessageFmt, curUID), curRemoteOKReason, apiv1.ConditionTrue)
+	finalConfig = cur
+	return
+}
+
+// StartSyncLoop launches the sync loop that watches the API server for configuration updates
+// The `client` passed to StartSyncLoop will replace the controller's current client
+func (cc *NodeConfigController) StartSyncLoop(client clientset.Interface, nodeName string) {
+	if client != nil {
+		cc.client = client
+		cc.nodeName = nodeName
+		infof("starting sync loop")
+
+		// start the configuration sync loop
+		go func() {
+			defer utilruntime.HandleCrash()
```

Comment:
context: `HandleCrash` doesn't `ReallyCrash` in the Kubelet; we should still call the panic handlers, but should actually crash if the config sync loop encounters a runtime error -- there is a better chance of a recovery if we crash.

I will make this change.